### DISCUSSION TITLE
Where to put my python file?

### DISCUSSION THREAD

**Discussion by cuber-1** [bitcraze/discussions#1707] [Q&A]
https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/user-guides/sbs_motion_commander/

I'm trying to make a python file to actually move the Crazyflie without any decks (I only have the AI deck, and will eventually try to use video feed to try and move it based on information it sees). In this documentation, it tells me how to make the python file, but not where to keep it. Also, how would I eventually run it?

**Comment by ToveRumar** [comment]
Hi!
Making the Crazyflie fly using a script and without any decks mounted will not work very well. 
It needs some sensory information to be able to stay stable in the air, unless controlled using a phone or gamepad controller. 

You can place the python file anywhere, however it is easiest if you copy an existing example file and modify it. The lib examples can be found in "your-path/crazyflie-lib-python/examples". 

You run it by calling
``python your-path/crazyflie-lib-python/examples/step-by-step/connect_log_param.py``

I hope this answered your question!

**Reply by cuber-1** [reply]
Would the AI Deck work? If I were to simply write a script that would make the Crazyflie fly for like 2 seconds and then stop, how would I do that exactly (from making the python file, to importing libraries)? Anything simple to get me started and understanding would be perfect. Thanks!

**Reply by ToveRumar** [reply]
Hi!

Making the Crazyflie fly stably with just an AI-deck might demand quite a lot of effort. I would suggest mounting a flow deck for stability. This will help you get stability so you can focus on other parts of your project, such as AI!

I can not provide any specific file for doing this using only the AI-deck but there are plenty of examples for making it fly with a positioning system or with a flow deck/multi ranger. 
This example will fly the drone in a pattern if it has a flow deck connected for example;  https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/step-by-step/sbs_motion_commander.py . From the examples you can create your own script that does what you want it to do!

Regarding the AI-deck it does not come with any pre-flashed AI-algorithms, so you need to create your own and train it. 
There are great examples here; https://github.com/bitcraze/aideck-gap8-examples/tree/master/examples where you can learn how to create your own algorithms and Neural networks! You can flash one of these examples and try them out to see how they work and what some of the possibilities are! 

I hope this helps you get started with your project!
